### PR TITLE
cadata param can (and PEM-encoded cadata must) be unicode under PY2

### PR DESCRIPTION
**Pull request by sfreilich**
The documentation for that parameter states: "The cadata object, if
present, is either an ASCII string of one or more PEM-encoded
certificates or a bytes-like object of DER-encoded certificates."
Similar wording is used for PY2 and PY3. But that's a bit of a
backporting bug in the documentation, since "ASCII string" and
"bytes-like object" sound like they could both be str under PY2. In
fact, if the type is str it's assumed to be DER-encoded and
PEM-encoded cadata must use the unicode type under PY2
(https://bugs.python.org/issue37079):
https://github.com/python/cpython/blob/2149a9ad/Modules/_ssl.c#L3011

Thus, these type declarations should use Union[Text, bytes, None]
instead of Union[str, bytes, None].

Also, _create_unverified_context exists in Python 2:
https://github.com/python/cpython/blob/2149a9ad/Lib/ssl.py#L448

### CODE REVIEW

File: stdlib/2and3/ssl.pyi [hunk 1]
```diff
@@ -52,9 +53,9 @@ if sys.version_info < (3,) or sys.version_info >= (3, 4):
     def create_default_context(purpose: Any = ..., *,
                                cafile: Optional[str] = ...,
                                capath: Optional[str] = ...,
-                               cadata: Union[str, bytes, None] = ...) -> SSLContext: ...
+                               cadata: Union[Text, bytes, None] = ...) -> SSLContext: ...
 
-if sys.version_info >= (3, 4):
+if sys.version_info < (3,) or sys.version_info >= (3, 4):
```
Comment (srittau) [inline stdlib/2and3/ssl.pyi]: Please just remove this version check. We only support Python 2.7 and 3.5+, so it is not necessary anymore.

Comment (sfreilich) [reply stdlib/2and3/ssl.pyi]: Done.

File: stdlib/2and3/ssl.pyi [hunk 2]
```diff
@@ -49,23 +49,21 @@ def wrap_socket(sock: socket.socket, keyfile: Optional[str] = ...,
                 ciphers: Optional[str] = ...) -> SSLSocket: ...
 
 
-if sys.version_info < (3,) or sys.version_info >= (3, 4):
```
Comment (sfreilich) [inline stdlib/2and3/ssl.pyi]: The diff chunking algorithm sure makes a hash of this one. It's a lot clearer with whitespace-only changes excluded from the diff.

**Comment by JelleZijlstra** [comment]
There is a merge conflict now.

**Comment by sfreilich** [comment]
> There is a merge conflict now.

Resolved, I think.

**Comment by JelleZijlstra** [comment]
CI is failing now though (haven't checked why).

**Comment by sfreilich** [comment]
Whitespace typo in the merge, missed when looking at the diff. Should be fixed.

**Comment by sfreilich** [comment]
This seems to be fixed, can someone merge this, please?

**Pull request merged**